src/dataframerow/dataframerow.jl

```diff
@@ -4,6 +4,15 @@ struct DataFrameRow{T <: AbstractDataFrame}
     row::Int
 end
 
+# this is defined here because it has to be after the DataFrameRow definition
```

Comment:
I don't really like this, it breaks the current organization of the indexing code. Can't you change the inclusion order of files so that `DataFrameRow` is defined before `DataFrame`? If that's not possible, we could move all type definitions to a separate file which would be included first.